### PR TITLE
ADM-415 FIX [frontend] [Config] - fix the bug in config page

### PR DESCRIPTION
**Pull request by spadio9587**
## Summary
Bugs in config page 
1.
Given change date option/start date/end date, 
the verify state for board/pipelineTool/sourceControl will be set to initial state( not verified)
2.
Given required date changed, if board section have already verified, 
when we choose more option in required data,
Then the pipelineTool component will show, board section verify state should still keep verified
3. 
Given filled all information in config page, and all component(board, pipelineTool, source control) verified, the next button was enabled, 
when click next button to go to metric page, after that click back button. then we will back to config page and all the data will stay and the state for all components are verified
4. 
Given filled all information in config page, and all component(board, pipelineTool, source control) verified, the next button was enabled, 
click next button to go to metric page, after that click back button. 
when change date option/start date/end date
the verify state for board/pipelineTool/sourceControl will be set to initial state( not verified)
5.
Given filled all information in config page, and all component(board, pipelineTool, source control) verified, the next button was enabled, 
click next button to go to metric page, after that click back button. 
when change date option/start date/end date
the verify state for board will be set to initial state( not verified)
board can not pass verify due to board type value is undefined

## Before

_Description_

Before

## After

_Description_

After

## Note

_Null_

### CODE REVIEW

**Review by mengna-tw — APPROVED** [review]
LGTM

File: frontend/src/store.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ export const store = configureStore({
     stepper: stepperReducer,
     config: configReducer,
     jiraVerifyResponse: jiraVerifyResponseReducer,
-    pipelineToolVerifyResponse: pipelineToolResponseReducer,
+    pipelineToolVerifyResponse: pipelineToolVerifyResponseReducer,
     sourceControlVerifyResponse: sourceControlVerifyResponseReducer,
```
Comment (LanxinWang) [inline frontend/src/store.ts:15]: could the 3 xxxResponseReducer to merged into one ResponseReducer ?

File: frontend/src/store.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ export const store = configureStore({
...
     saveMetricsSetting: saveMetricsSettingReducer,
   },
```
Comment (LanxinWang) [inline frontend/src/store.ts:17]: and I think board, pipeline tool, source control are similar,could we build a new card to refactor it ?

**Review by LanxinWang — APPROVED** [review]
LGTM

File: frontend/src/store.ts (hunk 1 above)
Comment (spadio9587) [reply frontend/src/store.ts:15]: Yep, i think that we can fix that later about refactor

File: frontend/src/store.ts (hunk 2 above)
Comment (spadio9587) [reply frontend/src/store.ts:17]: I think so about that!

**Pull request merged**